### ISSUE TITLE
Provide a way to build only libxkbcommon-x11 without rebuilding libxkbcommon

### ISSUE THREAD

**Issue by ryandesign** [xkbcommon/libxkbcommon#75]
The PACKAGING file says it is recommended to offer libxkbcommon and libxkbcommon-x11 as separate packages.

It's easy enough to build libxkbcommon without libxkbcommon-x11 by using `--disable-x11`. But I can't see how to do the reverse.

If I've already built and installed libxkbcommon without libxkbcommon-x11, and then later I want to install libxkbcommon-x11, I don't want to have to rebuild and reinstall all of libxkbcommon too.

There should be a way to build libxkbcommon-x11 only, against an already-installed libxkbcommon. If this capability already exists, please let me know; maybe the documentation needs improving. If it doesn't, it should be added.

**Comment by bluetech** [comment]
> I don't want to have to rebuild and reinstall all of libxkbcommon too

Why not?

**Comment by ryandesign** [comment]
Well for one thing, building things I don't need is a waste of CPU time.

For another, there's no clear method of installing *only* the files for libxkbcommon-x11. When configured with `--enable-x11`, `make install` installs the files for both libxkbcommon and libxkbcommon-x11. I had to resort to [traversing the installation directory and deleting all files whose names did not contain "x11"](https://github.com/macports/macports-ports/commit/ada3ede25c6d818bb5e9909fbeef8d099b57c52a). Surely a better solution could exist, such as a separate Makefile target I could use to build only libxkbcommon-x11 and a separate target that would install only the libxkbcommon-x11 files. Or else a configure switch that would control whether or not to build libxkbcommon, just as there is already the `--enable-x11` switch to control whether or not to build libxkbcommon-x11.

**Comment by fooishbar** [comment]
xkbcommon-x11 is useless without just plain xkbcommon.

The 'separate packages' advice refers to binary or built packages, not to source packages. The upstream advice is that you should build both at the same time, but allow users to choose whether or not to also install the optional X11 support. Debian/Ubuntu/Fedora/etc achieve this with separate binary packages built at the same time. Gentoo and similar distros achieve this with USE or feature flags.

**Comment by ryandesign** [comment]
Yes, I can understand why your build system might be designed this way, if you assume that package management systems produce multiple binary packages from a single source build. That's not how MacPorts works, however; each binary package is produced from a separate source build.

**Comment by fooishbar** [comment]
xkbcommon-x11's source depends on internal implementation details of xkbcommon itself that we are unwilling to make API/ABI.
